microsoft/botframework-components
# Command list template

### Purpose
*What is the context of this pull request? Why is it being done?*
Close #491 

### Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)*
* Apply design for Command List to a regex-recognizer bot

**Greeting**
![image](https://user-images.githubusercontent.com/43043272/106802794-46529e80-6618-11eb-9956-08bee1d81db8.png)
![image](https://user-images.githubusercontent.com/43043272/106803034-9467a200-6618-11eb-8d12-0cc97a3753ae.png)

**Hello**
![image](https://user-images.githubusercontent.com/43043272/106802757-3d61cd00-6618-11eb-9977-53255c40fbc6.png)
![image](https://user-images.githubusercontent.com/43043272/106803069-a21d2780-6618-11eb-80c6-e130d2684372.png)

**Help**
![image](https://user-images.githubusercontent.com/43043272/106802661-1acfb400-6618-11eb-8a13-f57aa5b2600d.png)
![image](https://user-images.githubusercontent.com/43043272/106803047-9b8eb000-6618-11eb-8918-856b296353c0.png)

**Unknown**
![image](https://user-images.githubusercontent.com/43043272/106802777-418dea80-6618-11eb-8d46-11c6774ae9d3.png)
![image](https://user-images.githubusercontent.com/43043272/106803089-a8ab9f00-6618-11eb-863d-8594f9485261.png)

### How to test
*Is this covered by existing tests or new ones? If no, why not?*
1. Pull down this branch.
2. Navigate to \generators\generator-command-list
3. Run `npm install`
4. Navigate to another directory to create your own bot and run `yo {path_to_\generators\generator-command-list\app\index.js {bot-name}`
5. Template should generate successfully and you can open in Composer to validate usability

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*
- [x] Depends on Bot Template Generator generator
- [x] Export this working bot into generator's /template directory
- [x] Provide README
- [x] Packaging
